### PR TITLE
The threshold in *ThresholdParameter can now be a Parameter.

### LINKED ISSUE #516: Control curve parameter based on another parameter instead of storage
I have need for an `IndexParameter` which compares a control curve (daily profile, etc) to another parameter's value rather than a `Storage` node volume. I don't think we have anything that does this currently?

### PR DESCRIPTION
**Pull request by snorfalorpagus**
Closes #516.

### CODE REVIEW

File: pywr/parameters/_thresholds.pyx [hunk 1]
```diff
@@ -42,9 +42,14 @@ cdef class AbstractThresholdParameter(IndexParameter):
     the previous day. In this case the predicate evaluates to True.
 
     """
-    def __init__(self, model, threshold, *args,  values=None, predicate=None, **kwargs):
+    def __init__(self, model, threshold, *args, values=None, predicate=None, **kwargs):
         super(AbstractThresholdParameter, self).__init__(model, *args, **kwargs)
-        self.threshold = threshold
+        threshold = load_parameter(model, threshold)
```
Comment (snorfalorpagus) [inline pywr/parameters/_thresholds.pyx]: This should be done in a `load` method.

File: pywr/parameters/_thresholds.pyx [hunk 2]
```diff
@@ -42,9 +42,14 @@ cdef class AbstractThresholdParameter(IndexParameter):
...
+        if isinstance(threshold, Parameter):
+            self._threshold_parameter = threshold
```
Comment (jetuk) [inline pywr/parameters/_thresholds.pyx]: Self should be made a parent of the threshold.

**Comment by snorfalorpagus** [comment]
I've addressed the two comments.

**Comment by jetuk** [comment]
This looks good. I think it is the right way around otherwise the test would fail.

**Pull request merged**